python/cugraph/cugraph/sampling/node2vec.py

```diff
@@ -0,0 +1,140 @@
+
+    Parameters
+    ----------
+    G : cuGraph.Graph or networkx.Graph
+        The graph can be either directed (DiGraph) or undirected (Graph).
+
+    start_vertices: int or list or cudf.Series
+        A single node or a list or a cudf.Series of nodes from which to run
+        the random walks
+
+    max_depth: int, optional
+        The maximum depth of the random walks
+
+    use_padding: bool, optional
+        If True, padded paths are returned else coalesced paths are returned
+
+    p: double, optional
+        Return factor, which represents the likelihood of backtracking to
+        a previous node in the walk. A higher value makes it less likely to
+        sample a previously visited node, while a lower value makes it more
+        likely to backtrack, making the walk "local"
+
+    q: double, optional
+        In-out factor, which represents the likelihood of visiting nodes
+        closer or further from the outgoing node. If q > 1, the random walk
+        is likelier to visit nodes closer to the outgoing node. If q < 1, the
+        random walk is likelier to visit nodes further from the outgoing node.
+
+    Returns
+    -------
+    vertex_paths : cudf.Series or cudf.DataFrame
+        Series containing the vertices of edges/paths in the random walk.
+
+    edge_weight_paths: cudf.Series
+        Series containing the edge weights of edges represented by the
+        returned vertex_paths
+
+    sizes: int or cudf.Series
+        The path size or sizes in case of coalesced paths.
+
+    Example
+    -------
+    >>> M = cudf.read_csv(datasets_path / 'karate.csv', delimiter=' ',
+    ...                   dtype=['int32', 'int32', 'float32'], header=None)
+    >>> G = cugraph.Graph()
+    >>> G.from_cudf_edgelist(M, source='0', destination='1', edge_attr='2')
+    >>> start_vertices = cudf.Series([0, 2])
+    >>> paths, weights, path_sizes = cugraph.node2vec(G, start_vertices, 3,
+    ...                                               True, 0.8, 0.5)
+
+    """
+    if (not isinstance(max_depth, int)) or (max_depth < 1):
+        raise ValueError("'max_depth' must be a positive integer")
+    if (not isinstance(use_padding, bool)):
+        raise ValueError("'use_padding' must be a bool")
+    if (p is None) or (p <= 0.0):
+        raise ValueError("'p' must be a positive double")
+    if (q is None) or (q <= 0.0):
+        raise ValueError("'q' must be a positive double")
+
+    G, _ = ensure_cugraph_obj_for_nx(G)
+
+    if isinstance(start_vertices, int):
+        start_vertices = [start_vertices]
+
+    if isinstance(start_vertices, list):
+        start_vertices = cudf.Series(start_vertices)
+
+    if G.renumbered is True:
+        if isinstance(start_vertices, cudf.DataFrame):
+            start_vertices = G.lookup_internal_vertex_id(
+                start_vertices, start_vertices.columns)
+        else:
+            start_vertices = G.lookup_internal_vertex_id(start_vertices)
+
+    srcs = G.edgelist.edgelist_df['src']
```

Comment:
This call will return the renumbered dataframe and will raise an error if the user didn't create a cugraph.Graph from a renumbered dataframe. You should probably first check if G is renumbered to avoid this

Reply:
Investigating this right now, the pylibcugraph wrapper does not work as intended if renumbered is False and returns an incorrect input. I added a fixme on line 124 describing this, and I believe this has to be resolved from the pylibcugraph or c layer